lib/matplotlib/tests/test_colorbar.py

```diff
@@ -712,6 +712,18 @@ def test_colorbar_label():
     assert cbar3.ax.get_xlabel() == 'horizontal cbar'
 
 
+@image_comparison(['colorbar_keeping_xlabel.png'], style='mpl20')
+def test_keeping_xlabel():
+    # github issue #23398 - xlabels being ignored in colorbar axis
+    rng = np.random.default_rng(19680801)
+    arr = rng.random((8, 8))
```

Comment:
We don't actually need a random image here, I think:
```suggestion
    arr = np.arange(9).reshape((3, 3))
```

Reply:
What is the advantage of np.arrange over random input here?

Reply:
This test isn't about the `pcolormesh`, so `arange` is simpler and (very slightly) less work.